apps/calendar/js/bridge.js

```diff
@@ -0,0 +1,118 @@
+define(function(require, exports) {
+'use strict';
+
+var EventEmitter2 = require('ext/eventemitter2');
+var co = require('ext/co');
+var core = require('core');
+var dayObserver = require('day_observer');
+
+/**
+ * Fetch all the data needed to display the busytime information on the event
+ * views based on the busytimeId
+ */
+exports.fetchRecord = function(busytimeId) {
+  return co(function *() {
+    var record = yield dayObserver.findAssociated(busytimeId);
+    var eventStore = core.storeFactory.get('Event');
+    var owners = yield eventStore.ownersOf(record.event);
+    var provider = core.providerFactory.get(owners.account.providerType);
+    var capabilities = yield provider.eventCapabilities(record.event);
+
+    record.calendar = owners.calendar;
+    record.account = owners.account;
+    record.capabilities = capabilities;
+
+    return record;
+  });
+};
+
+/**
+ * Fetch all the calendars from database and emits a new event every time the
+ * values changes.
+ *
+ * @returns {ClientStream}
+ */
+exports.listenCalendars = function() {
```

Comment:
So if I am a frontend like react, the way I will consume this is probably like

```js
var bridge = require('bridge');

component.getInitialState = function() {
  return { calendars: [] };
};

component.componentDidMount = function() {
  bridge.listenCalendars().on('data', calendars => this.setState({ calendars: calendars }));
};
```

Looks pretty awesome actually... what do you think about `observeCalendars()` instead of `listenCalendars()`?